lsp/source/server.mts

```diff
@@ -525,6 +525,13 @@ documents.onDidChangeContent(async ({ document }) => {
   console.log("onDidChangeContent", document.uri)
   changeQueue.add(document)
   scheduleExecuteQueue()
+
+  const sourcePath = documentToSourcePath(document)
+  assert(sourcePath)
+  const service = await ensureServiceForSourcePath(sourcePath)
+  if (!service) return
+
+  service.host.addOrUpdateDocument(document)
```

Comment:
It looks like this makes compilation synchronous again. We added the execution queue for a performance reason, and I'm worried that this will slow things down again. Maybe we can make completion await the completion of the queue instead?

Reply:
In fact, at the very beginning, I wrote it just as you said. I added a Promise to judge whether the queue has been updated. However, I encountered some problems.

I can't obtain the service at the queue. So, for convenience, I just wrote it here directly. If it is a performance issue, I will try to adjust it and see.

Reply:
If you could try again, that'd be great!

The code you added is already being done inside `updateDiagnosticsForDoc`, called from `updatePendingDiagnostics`, which is what's scheduled by the queue.

What I'm imagining is that some queue-related promise (which gets resolved when the queue runs) gets awaited *inside* `onCompletion` etc. Then it would hopefully be easy to still do everything else the same as we currently do...

Reply:
I will deal with it later. The time zone here is already four o'clock in the morning😂.